app/airflow/pyproject.toml

```diff
@@ -6,6 +6,10 @@ readme = "README.md"
 requires-python = ">=3.11"
 dependencies = [
     "apache-airflow>=2.10.5",
+    "apache-airflow-providers-amazon>=9.7.0",
+    "apache-airflow-providers-microsoft-azure>=12.3.1",
     "apache-airflow-providers-postgres>=6.1.1",
+    "openpyxl>=3.1.5",
+    "types-openpyxl>=3.1.5.20250506",
```

Comment:
Something to optimise in the future - the `types` are dev dependencies only, so can be grouped that way.